spec/timecop_spec.cr

```diff
@@ -1,122 +1,112 @@
-    Timecop.freeze(t) do
-      Timecop.return { raise "foobar" } rescue nil
-      true.should eq(t == Time.now)
+    it "with return unsets mock time" do
+      Timecop.freeze(Time.local)
+      Timecop.return
+      Timecop.frozen?.should be_false
     end
-  end
 
-  it "travel_keeps_time_moving" do
-    t = Time.new(2008, 10, 10, 10, 10, 10).to_local
-    now = Time.now
-    Timecop.travel(t) do
-      new_now = Time.now
-      true.should eq times_effectively_equal(new_now, t)
-      sleep(0.25)
-      false.should eq times_effectively_equal new_now, Time.now, 0.24
+    it "with block unsets mock time" do
+      Timecop.frozen?.should be_false
+      Timecop.freeze(Time.local) { }
+      Timecop.frozen?.should be_false
     end
-  end
 
-  it "doesnt break when utc_now is used" do
-    # Compile-time failure
-    now = Time.utc
-  end
+    it "restores previous time on raised exception within block" do
+      time = Time.local(2008, 10, 10, 10, 10, 10)
+      Timecop.freeze(time) do
+        Timecop.return { raise "foobar" } rescue nil
+        time.should eq(Time.local)
+      end
+    end
 
-  it "allows asking for the time with #now" do
-    time = Time.new
-    Timecop.freeze(time) do |t|
-      sleep 0.25
-      Time.now.should eq t
+    it "allows asking for the time with .local" do
+      time = Time.local
+      Timecop.freeze(time) do |t|
+        sleep(250.milliseconds)
+        Time.local.should eq(t)
+      end
     end
-  end
 
-  it "allows asking for the time with #utc" do
-    time = Time.new
-    Timecop.freeze(time) do |t|
-      sleep 0.25
-      Time.utc.should eq t
+    it "allows asking for the time with .utc" do
+      time = Time.utc
+      Timecop.freeze(time) do |t|
+        sleep(250.milliseconds)
+        Time.utc.should eq(t)
+      end
     end
   end
 
-  it "allows asking for the time with #local" do
-    time = Time.new
-    Timecop.freeze(time) do |t|
-      sleep 0.25
-      Time.local.should eq t
+  context ".scale" do
+    it "keeps time moving at an accelerated rate" do
+      time = Time.local(2008, 10, 10, 10, 10, 10)
+      Timecop.scale(time, 4) do
+        start = Time.local
+        start.should be_close(time, 1.second)
+        sleep(250.milliseconds)
+        (start + 1.second).should be_close(Time.local, 250.milliseconds)
+      end
+    end
+
+    it "returns `now` if no block given" do
+      time = Time.local(2008, 10, 10, 10, 10, 10)
+      time.should be_close(Timecop.scale(time, 4), 1.second)
     end
   end
 
-  pending "travel_does_not_reduce_precision_of_datetime" do
-    Timecop.travel(Time.new(2014, 1, 1, 0, 0, 0))
-    true.should eq (Time.now != Time.now)
+  context ".travel" do
+    it "keeps time moving" do
+      time = Time.local(2008, 10, 10, 10, 10, 10)
+      Timecop.travel(time) do
+        start = Time.local
+        start.should be_close(time, 1.second)
+        sleep(250.milliseconds)
+        start.should_not be_close(Time.local, 240.milliseconds)
```

Comment:
250ms seems like a lot of tolerance for these tests. Since we are manipulating the time clock, maybe we should be testing with more accuracy?

Reply:
It was like that before, I don't mind changing it but IMO it doesn't seem to be super important.